### PR TITLE
Add 3.11 pipeline

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.11.groovy [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            "http://koji.katello.org/releases/yum/katello-3.11/${repo}/${dist}/x86_64/",
+            "-l ${dist}-foreman-1.21",
+            "-l ${dist}-foreman-plugins-1.21",
+            "-l ${dist}-foreman-rails-1.21",
+            "-l ${dist}-base",
+            "-l ${dist}-updates",
+            "-l ${dist}-epel",
+            "-l ${dist}-extras",
+            "-l ${dist}-scl",
+            "-l ${dist}-scl-sclo",
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.11.groovy]: sclo is no longer needed and can be removed (AFAIK it doesn't even exist in the .conf file)

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.11.groovy [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        stage('Katello Repoclosure') {
+            agent { label 'el' }
+
+            steps {
+
+                repoclosure('katello', 'el7')
+
+            }
+        }
+        stage('Install Test') {
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.11.groovy]: Install and upgrade Test

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.11.groovy [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            parallel {
+                stage('Install test') {
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.11.groovy]: both stages are missing an "allways post" step with `set_job_build_description("${env.cico_job_name}")`  in them (but we also don't have that in other versioned katello jobs - intentional?)

Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.11.groovy]: I added it to nightly for debugging originally and never propagated it out to versioned pipelines. I will at least add it here.

**Comment by ehelms** [comment]
I suppose these jobs are getting a bit formulaic now. Updated with all comments to get this available for RC1.

**Comment by ehelms** [comment]
@ekohl @evgeni please re-review

**Pull request merged**